lua/entities/gmod_wire_expression2/core/custom/wiring.lua

```diff
@@ -18,8 +18,8 @@ e2function number entity:createWire(entity ent2, string inputname, string output
 
 	local trigger = self.entity.trigger
 	self.entity.trigger = { false, {} } -- So the wire creation doesn't execute the E2 immediately because an input changed
-	WireLib.Link_Start(self.player:UniqueID(), this, this:WorldToLocal(this:GetPos()), inputname, "cable/rope", Vector(255,255,255), 0)
-	WireLib.Link_End(self.player:UniqueID(), ent2, ent2:WorldToLocal(ent2:GetPos()), outputname, self.player)
+	WireLib.Link_Start(1, this, this:WorldToLocal(this:GetPos()), inputname, "cable/rope", Vector(255,255,255), 0) -- Divran, I'm trusting you on using 1 as index here instead
```

Comment:
It is because of an ancient code, but it was refactored since. And Divran told me, it would be fine to use `1` or any value. Since `Link_End` is called immediately after `Link_Start`, it will always create a link correctly.
If we could get rid of this argument somehow, without breaking 3rd-party addons, then we wouldn't even need a comment here.

Reply:
> If we could get rid of this argument somehow, without breaking 3rd-party addons, then we wouldn't even need a comment here.

That's what I'm suggesting: we can create an API which doesn't need to take this argument. And for now it can even just be implemented in terms of the old API. Here:

```lua
--- Create a wire directly from output_ent to input_ent with the given parameters.
--- `player` may be nil, NULL or game.GetWorld if this wire isn't being created for a player.
--- `width` may be nil, which gives an invisible wire.
--- `color` may be nil, which gives a white wire.
--- `material` may be nil, which gives a wire with an unspecified but valid material.
function WireLib.Link(player, input_ent, input_localpos, input_name,
                      output_ent, output_localpos, output_name,
                      width, color, material)
    player = IsValid(player) and player or nil
    material = material or "arrowire/arrowire2"
    color = color or Color(255, 255, 255, 255)
    width = width or 0
    -- The actual value of this `id` parameter doesn't matter, but it needs to never be
    -- used by anybody else calling WireLib.Link_{Start,End}. This will be used as a
    -- key into a global table of in-progress links. Other users of the incremental API
    -- typically pass in the string returned by `player:UniqueID()`, so we pass in a
    -- function just to avoid ever colliding with them.
    local id = WireLib.Link
    WireLib.Link_Start(id, input_ent, input_localpos, input_name, material, color, width)
    WireLib.Link_End(id, output_ent, output_localpos, output_name, player)
end
```